pkg/graphql/federation/required_field_extractor.go

```diff
@@ -0,0 +1,130 @@
+package federation
+
+import (
+	"strings"
+
+	"github.com/jensneuse/graphql-go-tools/pkg/ast"
+	"github.com/jensneuse/graphql-go-tools/pkg/engine/plan"
+)
+
+// requiredFieldExtractor
+type requiredFieldExtractor struct {
```

Comment:
I see this one also in another package. It's more low-level and moving it to a proper package could make it more reusable.